src/AasxServerDB/Entities/SMESet.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Extensions;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace AasxServerDB.Entities
+{
+    public class SMESet
+    {
+        public int Id { get; set; }
+
+        [ForeignKey("SMSet")]
+        public int SMId { get; set; }
+        public virtual SMSet SMSet { get; set; }
+
+        public int? ParentSMEId { get; set; }
+        public virtual SMESet? ParentSME { get; set; }
+
+        public string? SMEType { get; set; }
+        public string? ValueType { get; set; }
+        public string? SemanticId { get; set; }
+        public string? IdShort { get; set; }
+
+        public virtual ICollection<IValueSet> IValueSets { get; } = new List<IValueSet>();
+        public virtual ICollection<DValueSet> DValueSets { get; } = new List<DValueSet>();
+        public virtual ICollection<SValueSet> SValueSets { get; } = new List<SValueSet>();
+
+        public string getValue()
+        {
+            using (AasContext db = new AasContext())
+            {
+                switch (ValueType)
+                {
+                    case "S":
+                        var ls = db.SValueSets.Where(s => s.SMEId == Id).Select(s => s.Value).ToList();
+                        if (ls.Count != 0)
+                            return ls.First().ToString();
+                        break;
+                    case "I":
+                        var li = db.IValueSets.Where(s => s.SMEId == Id).Select(s => s.Value).ToList();
+                        if (li.Count != 0)
+                            return li.First().ToString();
+                        break;
+                    case "F":
+                        var ld = db.DValueSets.Where(s => s.SMEId == Id).Select(s => s.Value).ToList();
+                        if (ld.Count != 0)
+                            return ld.First().ToString();
+                        break;
+                }
+            }
+            return string.Empty;
+        }
+
+        public List<string> getMLPValue()
+        {
+            var list = new List<string>();
+            if (SMEType == "MLP")
+            {
+                using (AasContext db = new AasContext())
+                {
+                    var mlpValueSetList = db.SValueSets.Where(s => s.SMEId == Id).ToList();
+                    foreach (var mlpValue in mlpValueSetList)
+                    {
+                        list.Add(mlpValue.Annotation);
+                        list.Add(mlpValue.Value);
+                    }
+                    return list;
+                }
+            }
+            return new List<string>();
+        }
+
+        public static List<SValueSet> getValueList(List<SMESet> smesets)
+        {
+            var smeIds = smesets.OrderBy(s => s.Id).Select(s => s.Id).ToList();
+            long first = smeIds.First();
+            long last = smeIds.Last();
+            List<SValueSet> valueList = null;
+            using (AasContext db = new AasContext())

```

Comment:
Could make this a bit saver and easier readable

```suggestion
            using (AasContext db = new AasContext())
{
    var watch = System.Diagnostics.Stopwatch.StartNew();

    var query = @"
        SELECT * FROM SValueSets WHERE ParentSMEId >= {0} AND ParentSMEId <= {1}
        UNION
        SELECT * FROM IValueSets WHERE ParentSMEId >= {0} AND ParentSMEId <= {1}
        UNION
        SELECT * FROM DValueSets WHERE ParentSMEId >= {0} AND ParentSMEId <= {1}
    ";

    valueList = await db.SValueSets.FromSqlRaw(query, first, last)
        .Where(v => smeIds.Contains(v.SMEId))
        .OrderBy(v => v.SMEId)
        .ToListAsync();

    watch.Stop();
    Console.WriteLine($"Getting the value list took this time: {watch.ElapsedMilliseconds} ms");
}

```